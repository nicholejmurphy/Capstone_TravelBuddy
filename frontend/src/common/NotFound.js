import React from "react";

function NotFound() {
  return (
    <div>
      <div className="NotFound bg-dark text-light p-4 rounded shadow">
        <h2>404 Error.</h2>
        <p className="lead">
          The requested URL was not found on this server ðŸ™ƒ
        </p>
      </div>
    </div>
  );
}

export default NotFound;
