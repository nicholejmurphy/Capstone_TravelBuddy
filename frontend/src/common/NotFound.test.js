import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import NotFound from "./NotFound";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <NotFound />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <NotFound />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
