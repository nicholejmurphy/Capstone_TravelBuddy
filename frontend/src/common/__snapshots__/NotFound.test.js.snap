// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="NotFound bg-dark text-light p-4 rounded shadow"
    >
      <h2>
        404 Error.
      </h2>
      <p
        class="lead"
      >
        The requested URL was not found on this server ðŸ™ƒ
      </p>
    </div>
  </div>
</DocumentFragment>
`;
